gradle/test-launcher.gradle

```diff
@@ -0,0 +1,13 @@
+tasks.withType(Test) {
+    javaLauncher = javaToolchains.launcherFor {
+        languageVersion = providers
+            .gradleProperty("mockito.test.java")
+            .map {
+                if (it == "auto") {
+                    JavaLanguageVersion.of(JavaVersion.current().majorVersion)
```

Comment:
Shouldn't we now also remove the target and source compatibility definitions, now that we use toolchains?

Reply:
The toolchains are only for tests for now. I didn't change the compilation (`compileJava` / `compileTestJava` works as before (always on 11)).

Reply:
<sup>1</sup>: `java-version: matrix.java -> 17` [here in `build:` job](https://github.com/mockito/mockito/pull/3064/files#diff-b803fcb7f17ed9235f1e5cb1fcd2f5d3b2838429d4368ae4c57ce4436577f03fR53)
<sup>2</sup>: `java-version: 11 -> 17` [here in `release:` job](https://github.com/mockito/mockito/pull/3064/files#diff-b803fcb7f17ed9235f1e5cb1fcd2f5d3b2838429d4368ae4c57ce4436577f03fR210)
<sup>3</sup>: implicit Gradle Java runtime to explicit toolchain [here](https://github.com/mockito/mockito/pull/3064/files#diff-b803fcb7f17ed9235f1e5cb1fcd2f5d3b2838429d4368ae4c57ce4436577f03fR74)
Java 11 bytecode is forced by targetCompatibility.

Before:
 * build: Test matrix Java 11<sup>1</sup> producing Java 11 bytecode running on Java 11<sup>3</sup>
 * build: Test matrix Java 17<sup>1</sup> producing Java 11 bytecode running on Java 17<sup>3</sup>
 * release: Gradle on Java 11<sup>2</sup> producing Java 11 bytecode.

After:
 * build: Test matrix Java 17<sup>1</sup> producing Java 11 bytecode running on Java 11<sup>3</sup>
 * build: Test matrix Java 17<sup>1</sup> producing Java 11 bytecode running on Java 17<sup>3</sup>
 * release Gradle on Java 17<sup>2</sup> producing Java 11 bytecode.

The key here is that the built bytecode is always the same, and is tested on different runtimes. Same as user's envs.

I think unless Java 21+ LTS [deprecates target=11](https://inside.java/2022/09/26/quality-heads-up/), we can keep using this pattern. If deprecation happens we can increase minimum, or force a toolchain, but that would remove optimizations/fixes of javac in Java 12-17.

Reply:
Yes this looks fine to me. Thanks for the explanation!